core/src/main/java/io/micronaut/core/type/RuntimeTypeInformation.java

```diff
@@ -39,29 +40,23 @@
  */
 @Internal
 final class RuntimeTypeInformation {
-    private static final Map<Class<?>, Argument<?>> WRAPPER_TO_TYPE = new HashMap<>(3);
-    private static final Collection<TypeInformationProvider> TYPE_INFORMATION_PROVIDERS;
 
-    static {
-        WRAPPER_TO_TYPE.put(OptionalDouble.class, Argument.DOUBLE);
-        WRAPPER_TO_TYPE.put(OptionalLong.class, Argument.LONG);
-        WRAPPER_TO_TYPE.put(OptionalInt.class, Argument.INT);
-        final ServiceLoader<TypeInformationProvider> loader = ServiceLoader.load(TypeInformationProvider.class);
-        List<TypeInformationProvider> informationProviders = new ArrayList<>(2);
-        for (TypeInformationProvider informationProvider : loader) {
-            informationProviders.add(informationProvider);
-        }
-
-        TYPE_INFORMATION_PROVIDERS = Collections.unmodifiableList(informationProviders);
+    private static boolean isJavaBasicTypeAndNotReactiveAndNotWrapper(Class<?> type) {
+        // Not of them are reactive or wrappers
+        return ClassUtils.isJavaBasicType(type);
     }
 
     /**
      * Returns whether the annotation metadata specifies the type as single.
+     * @param type The return type
      * @param annotationMetadata The annotation metadata provider
      * @return True if does
      */
-    static boolean isSpecifiedSingle(AnnotationMetadataProvider annotationMetadata) {
-        for (TypeInformationProvider provider : TYPE_INFORMATION_PROVIDERS) {
+    static boolean isSpecifiedSingle(Class<?> type, AnnotationMetadataProvider annotationMetadata) {
```

Comment:
Added an extra check to avoid reactive check when the type is a basic Java type.